includes/class-wc-stripe-logger.php

```diff
@@ -21,41 +21,89 @@ class WC_Stripe_Logger {
 	 * @version 4.0.0
 	 */
 	public static function log( $message, $start_time = null, $end_time = null ) {
-		if ( ! class_exists( 'WC_Logger' ) ) {
+		if ( ! self::can_log() ) {
 			return;
 		}
 
-		if ( apply_filters( 'wc_stripe_logging', true, $message ) ) {
-			if ( empty( self::$logger ) ) {
-				self::$logger = wc_get_logger();
-			}
+		if ( ! apply_filters( 'wc_stripe_logging', true, $message ) ) {
+			return;
+		}
 
-			$settings = get_option( 'woocommerce_stripe_settings' );
+		if ( empty( self::$logger ) ) {
+			self::$logger = wc_get_logger();
+		}
 
-			if ( empty( $settings ) || isset( $settings['logging'] ) && 'yes' !== $settings['logging'] ) {
-				return;
-			}
+		$log_entry  = "\n" . '====Stripe Version: ' . WC_STRIPE_VERSION . '====' . "\n";
+		$log_entry .= '====Stripe Plugin API Version: ' . WC_Stripe_API::STRIPE_API_VERSION . '====' . "\n";
 
-			if ( ! is_null( $start_time ) ) {
+		if ( ! is_null( $start_time ) ) {
+			$formatted_start_time = date_i18n( get_option( 'date_format' ) . ' g:ia', $start_time );
+			$end_time             = is_null( $end_time ) ? current_time( 'timestamp' ) : $end_time;
+			$formatted_end_time   = date_i18n( get_option( 'date_format' ) . ' g:ia', $end_time );
+			$elapsed_time         = round( abs( $end_time - $start_time ) / 60, 2 );
 
-				$formatted_start_time = date_i18n( get_option( 'date_format' ) . ' g:ia', $start_time );
-				$end_time             = is_null( $end_time ) ? current_time( 'timestamp' ) : $end_time;
-				$formatted_end_time   = date_i18n( get_option( 'date_format' ) . ' g:ia', $end_time );
-				$elapsed_time         = round( abs( $end_time - $start_time ) / 60, 2 );
+			$log_entry .= '====Start Log ' . $formatted_start_time . '====' . "\n" . $message . "\n";
+			$log_entry .= '====End Log ' . $formatted_end_time . ' (' . $elapsed_time . ')====' . "\n\n";
 
-				$log_entry  = "\n" . '====Stripe Version: ' . WC_STRIPE_VERSION . '====' . "\n";
-				$log_entry .= '====Stripe Plugin API Version: ' . WC_Stripe_API::STRIPE_API_VERSION . '====' . "\n";
-				$log_entry .= '====Start Log ' . $formatted_start_time . '====' . "\n" . $message . "\n";
-				$log_entry .= '====End Log ' . $formatted_end_time . ' (' . $elapsed_time . ')====' . "\n\n";
+		} else {
+			$log_entry .= '====Start Log====' . "\n" . $message . "\n" . '====End Log====' . "\n\n";
+		}
 
-			} else {
-				$log_entry  = "\n" . '====Stripe Version: ' . WC_STRIPE_VERSION . '====' . "\n";
-				$log_entry .= '====Stripe Plugin API Version: ' . WC_Stripe_API::STRIPE_API_VERSION . '====' . "\n";
-				$log_entry .= '====Start Log====' . "\n" . $message . "\n" . '====End Log====' . "\n\n";
+		self::$logger->debug( $log_entry, [ 'source' => self::WC_LOG_FILENAME ] );
+	}
 
-			}
+	/**
+	 * Creates a log entry of type error.
+	 *
+	 * @param string $message To send to the log file.
+	 * @return void
+	 */
+	public static function error( $message ) {
+		if ( ! self::can_log() ) {
```

Comment:
I intentionally omitted including the `====Stripe Version====` , `====Start Log====`, and other wrappers we add in `self::log()`.

Having information about the environment variables is helpful, but having them on each log entry pollutes the logs with redundant information. We could have a separate method to log the environment information and call it when relevant, for example. But I'm simply omitting it for these new methods in this PR to keep it small and because we're already including it in each call to `self::log()`.

Reply:
I agree, they add very little context... I think that the dates and error type are enough.